src/test/java/com/github/princesslana/smalld/TestHeartbeat.java

```diff
@@ -13,37 +13,40 @@
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.Mock;
 import org.mockito.Mockito;
+import org.mockito.Spy;
 import org.mockito.junit.jupiter.MockitoExtension;
 
 @ExtendWith(MockitoExtension.class)
 class TestHeartbeat {
 
   private Heartbeat subject;
 
-  private MockSmallD smalld;
+  @Spy private MockSmallD smalld;
```

Comment:
I think we want to avoid having to Spy on a Mock that is within our code base.

If checking certain functionality is useful for tests, we should look to support that in MockSmallD.

Reply:
My issue with this is you can't predict which set of this type of functionality would be useful. Like, if I add `assertCalledReconnectOnce`, shouldn't there be a similar one for other methods as well? A spy would do all of this with no effort. If we want a generic approach, something like `assertCalled(Callable function, int times)` or w/e, then its just out of this PR's scope.

Did you have a simpler idea in mind?

Reply:
Or actually, you just want to include `assertCalledReconnectOnce`? :thinking: